types/json_path_expr.go

```diff
@@ -50,34 +50,84 @@ import (
 		select json_extract('{"a": "b", "c": [1, "2"]}', '$.*') -> ["b", [1, "2"]]
 */
 
+type jsonPathArrayIndex struct {
+	fromLast bool
+	index    int
```

Comment:
Can we use negative values to express `fromLast`?

Reply:
No, negative values are not allowed. 

It's still a feature request for MySQL: https://bugs.mysql.com/bug.php?id=99238. I have no idea why MySQL chooses to use `last - xxx` rather than `- xxx` (the only difference is that `-1` means last, and `last - 1` means the element before last).

Reply:
> No, negative values are not allowed.

Yeah, I mean can we use negative values internally so that we don't need fromLast. If the only difference is an offset by one, sure you can use `-xxx` for `- xxx`, and `-xxx-1` for `last - xxx`.

Reply:
Good!

Reply:
Done.